File: Python/film.py
class Film:
    """Class Model Film untuk menyimpan data objek film."""

    def __init__(self, id_film, judul, genre, harga):
        # Atribut private menggunakan prefix dunder (__), bentuk enkapsulasi
        self.__id = id_film
        self.__judul = judul
        self.__genre = genre
        self.__harga = harga

    # --- Getters ---
    def get_id(self):
        return self.__id

    def get_judul(self):
        return self.__judul

    def get_genre(self):
        return self.__genre

    def get_harga(self):
        return self.__harga

    # --- Setters ---
    def set_judul(self, judul):
        self.__judul = judul

    def set_genre(self, genre):
        self.__genre = genre

    def set_harga(self, harga):
        self.__harga = harga

File: PHP/film.php
<?php
// Class model untuk menyimpan struktur data objek film
class Film {
    private $id;
    private $judul;
    private $genre;
    private $harga;
    private $gambar; // Nama file gambar (misal: poster.jpg)

    public function __construct($id, $judul, $genre, $harga, $gambar) {
        $this->id = $id;
        $this->judul = $judul;
        $this->genre = $genre;
        $this->harga = $harga;
        $this->gambar = $gambar;
    }

    // Getter
    public function getId() { return $this->id; }
    public function getJudul() { return $this->judul; }
    public function getGenre() { return $this->genre; }
    public function getHarga() { return $this->harga; }
    public function getGambar() { return $this->gambar; }

    // Setter
    public function setJudul($judul) { $this->judul = $judul; }
    public function setGenre($genre) { $this->genre = $genre; }
    public function setHarga($harga) { $this->harga = $harga; }
    public function setGambar($gambar) { $this->gambar = $gambar; }
}
?>